src/dune_engine/build_system.ml

```diff
@@ -2389,6 +2389,39 @@ let is_target file =
   let+ targets = targets_of ~dir:(Path.parent_exn file) in
   Path.Set.mem targets file
 
+(* Evaluate a rule without building the action dependencies *)
+module Eval_action_builder = Action_builder.Make_exec (struct
+  type fact = unit
+
+  let merge_facts = Dep.Set.union
+
+  let read_file p ~f =
+    let+ _digest = build_file p in
+    f p
+
+  let register_action_deps deps = Memo.Build.return deps
+
+  let register_action_dep_pred g =
+    let+ ps = Pred.eval g in
+    (ps, ())
+
+  let file_exists = file_exists
+
+  let alias_exists = Load_rules.alias_exists
+
+  let execute_action ~observing_facts:_ _act =
```

Comment:
The fact that doing nothing is a valid implementation of `execute_action` makes me think that the name of this function can be improved. Perhaps, `register_action` would be more appropriate, similar to `register_action_deps`? Another alternative is `handle_action` and `handle_action_deps`.

Reply:
That's definitely wrong. I blindly copied the code from the `For_command_line` module, but clearly we can't apply the same policy there. I changed the code.

Reply:
Actually, I ended up just changing the comments to explain better what `eval_build_request` does.

Reply:
I ended up taking a different approach and reverting changes to `Build_system`.